src/execution_plan/ops/op.h

```diff
@@ -60,26 +60,27 @@ typedef enum {
 } OpResult;
 
 struct OpBase;
+struct ExecutionPlan;
 
 typedef void (*fpFree)(struct OpBase *);
 typedef OpResult(*fpInit)(struct OpBase *);
 typedef Record(*fpConsume)(struct OpBase *);
 typedef OpResult(*fpReset)(struct OpBase *);
 typedef int (*fpToString)(const struct OpBase *, char *, uint);
+typedef struct OpBase *(*fpClone)(const struct ExecutionPlan *, const struct OpBase *);
```

Comment:
Why do you want to use the `ExecutionPlan` as an argument here? We can just access `op->plan`, no?